ui/src/components/Secrets.vue

```diff
@@ -517,6 +577,75 @@ function removeSecret(counter) {
     secrets.value[0] = { key: "", value: "", file: [] }
   }
 }
+
+function generateSecureRandomString(length, includeUppercase, includeLowercase, includeNumbers, includeSpecial) {
+  const classes = {
+    upperCase: { chars: "ABCDEFGHIJKLMNOPQRSTUVWXYZ", enabled: includeUppercase },
+    lowerCase: { chars: "abcdefghijklmnopqrstuvwxyz", enabled: includeLowercase },
+    digits: { chars: "0123456789", enabled: includeNumbers },
+    special: { chars: specialCharacters.value, enabled: includeSpecial }
+  };
+
+  const chars = Object.values(classes)
+    .filter(v => v.enabled)
+    .map(v => v.chars)
+    .join('');
+
+  if (chars.length === 0) {
+    setErrorMessage('At least one character type must be selected to create a random string.');
+    return '';
+  }
+
+  let initialResult = '';
+
+  Object.values(classes).forEach(v => {
+    if (v.enabled) {
+      initialResult += getRandomCharacter(v.chars);
+    }
+  });
+  for (let i = initialResult.length; i < length; i++) {
+    initialResult += getRandomCharacter(chars);
+  }
+  const result = shuffleString(initialResult);
+  return result;
+}
+
+function shuffleString(str) {
+  const array = str.split('');
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+  return array.join('');
+}
+
+function getRandomCharacter(charSet) {
+    if (charSet.length === 0) {
+        throw new Error("Character set cannot be empty");
+    }
+    const randomValues = new Uint32Array(1);
+    window.crypto.getRandomValues(randomValues);
+    // Generate a random index within the bounds of charSet length
+    // By dividing the random number by 0xFFFFFFFF + 1, you normalize it to a value between 0 and 1.
+    // Multiplying this normalized value by charSet.length gives you a floating-point number that scales to the desired range.
+    const randomIndex = Math.floor(randomValues[0] / (0xFFFFFFFF + 1) * charSet.length);
+  
+    return charSet[randomIndex];
+}
+
+
+
+function generateRandomString(index) {
+  const length = passwordLength.value;
+  const config = {
```

Comment:
I would rather like to see such a hash to be passed into the function generateSecureRandomString, than keeping the long parameter list, as is done now.